src/tasks/dump.ts

```diff
@@ -0,0 +1,567 @@
+  );
+  if (sameTokenIndex !== -1) {
+    dumpedTokens.splice(sameTokenIndex, 1);
+    if (hasCustomReceiver) {
+      transferToReceiver = {
+        token: toToken,
+        amount: await balanceOf(toToken, user),
+      };
+    }
+  }
+
+  const computedInstructions: (DumpInstruction | null)[] = (
+    await promiseAllWithRateLimit(
+      dumpedTokens.map((tokenAddress) => async ({ consoleWarn }) => {
+        const token = await tokenDetails(tokenAddress, hre);
+        const [balance, approvedAmount]: [
+          BigNumber,
+          BigNumber,
+        ] = await Promise.all([
+          token.contract.balanceOf(user).then(BigNumber.from),
+          token.contract.allowance(user, allowanceManager).then(BigNumber.from),
+        ]);
+        if (balance.isZero()) {
+          consoleWarn(
+            `Dump request skipped for token ${displayName(
+              token,
+            )}. No balance for that token is available.`,
+          );
+          return null;
+        }
+        const sellToken = token.address;
+        const buyToken = isNativeToken(toToken)
+          ? WRAPPED_NATIVE_TOKEN_ADDRESS[network] // todo: replace WETH address with BUY_ETH_ADDRESS when services support ETH estimates
+          : toToken.address;
+        const kind = OrderKind.SELL;
+        const fee = await getFee({
+          sellToken,
+          buyToken,
+          kind,
+          amount: balance,
+          network,
+        });
+        const amount = balance.sub(fee);
+        if (amount.lte(constants.Zero)) {
+          consoleWarn(
+            `Dump request skipped for token ${displayName(
+              token,
+            )}. The trading fee is larger than the dumped amount.`,
+          );
+          return null;
+        }
+        const approxBalance = Number(balance.toString());
+        const approxFee = Number(fee.toString());
+        const feePercent = (100 * approxFee) / approxBalance;
+        if (feePercent > maxFeePercent) {
+          consoleWarn(
+            `Dump request skipped for token ${displayName(
+              token,
+            )}. The trading fee is too large compared to the balance (${feePercent.toFixed(
+              2,
+            )}%).`,
+          );
+          return null;
+        }
+        const receivedAmount = await estimateTradeAmount({
+          sellToken,
+          buyToken,
+          amount,
+          kind,
+          network,
+        });
+        return {
+          token,
+          balance,
+          amount,
+          approvedAmount,
+          receivedAmount,
+          fee,
+        };
+      }),
+      { rateLimit: 10 },
+    )
+  ).filter((inst) => inst !== null);
+  // note: null entries have already been filtered out
+  const instructions = computedInstructions as DumpInstruction[];
+  instructions.sort((lhs, rhs) =>
+    lhs.receivedAmount.eq(rhs.receivedAmount)
+      ? 0
+      : lhs.receivedAmount.lt(rhs.receivedAmount)
+      ? -1
+      : 1,
```

Comment:
This is an interesting way to sort, although I wonder if there is just some way to specify a sort key (i.e the receivedAmount) and not explicitly the comparator function itself. In python this would be accomplished with a lambda function (aka closure?)

Sorted(my_list, key=entry: entry.receivedAmount)